classes/QueryCondition.cls

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copied and slightly modified from https://www.jamessimone.net/blog/joys-of-apex/repository-pattern/
+ */
+public with sharing class QueryCondition implements Comparable {
+    public enum Operator {
+        EQUALS,
+        NOT_EQUALS,
+        LESS_THAN,
+        LESS_THAN_OR_EQUAL,
+        GREATER_THAN,
+        GREATER_THAN_OR_EQUAL
+    }
+
+    private final String fieldName;
+    private final Operator operator;
+    private final List<Object> values;
+
+    private static Boolean isSet = false;
```

Comment:
isSet shouldn't be static here. There's the chance multiple queries are run on the same sobject in the same test method, and this value will persist unless it's reset on every instance.

Reply:
this was copied from James' blog and I hadn't run into it yet, good catch!